workflows/src/test/kotlin/com/template/ResponseTests.kt

```diff
@@ -0,0 +1,121 @@
+package com.template
+
+import com.template.flows.ApproveAppointmentRequest
+import com.template.flows.CreateAppointmentDate
+import com.template.flows.CreateAppointmentRequest
+import com.template.flows.DenyAppointmentRequest
+import com.template.schemas.AppointmentDateSchemaV1
+import com.template.states.AppointmentRequest
+import net.corda.testing.node.*
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.concurrent.Future
+import net.corda.core.node.services.vault.QueryCriteria
+import net.corda.core.transactions.SignedTransaction
+import com.template.states.AvailableAppointmentDate
+import net.corda.core.node.services.Vault
+import net.corda.core.node.services.vault.Builder.equal
+import net.corda.core.utilities.getOrThrow
+import net.corda.testing.common.internal.testNetworkParameters
+import java.util.*
+import kotlin.test.assertEquals
+
+
+class ResponseTests {
+    private lateinit var network: MockNetwork
+    private lateinit var doctor: StartedMockNode
+    private lateinit var alice: StartedMockNode
+    private lateinit var bob: StartedMockNode
+
+    @Before
+    fun setup() {
+        val myNetworkParameters = testNetworkParameters(minimumPlatformVersion = 4)
+        network = MockNetwork(MockNetworkParameters(cordappsForAllNodes = listOf(
+            TestCordapp.findCordapp("com.template.contracts"),
+            TestCordapp.findCordapp("com.template.flows")
+        ),
+            networkParameters = myNetworkParameters))
+        doctor = network.createPartyNode()
+        alice = network.createPartyNode()
+        bob = network.createPartyNode()
+        network.runNetwork()
+    }
+
+    @After
+    fun tearDown() {
+        network.stopNodes()
+    }
+    @Test
+    fun approveRequest() {
+        val availableDateFlow = CreateAppointmentDate(alice.info.legalIdentities[0], bob.info.legalIdentities[0], "06-07-2021")
```

Comment:
Could you please add more tests? For example:
- there are a few available dates and you ensure that only the appointment date is consumed
- if you provide a date which doesn't match a request, the flow fails